build.sbt

```diff
@@ -290,6 +290,47 @@ lazy val munitScalacheck = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .jsConfigure(sharedJSConfigure)
   .jsSettings(sharedJSSettings)
 
+lazy val munitCore = crossProject(JSPlatform, JVMPlatform, NativePlatform)
+  .in(file("munit-core"))
+  .settings(
+    moduleName := "munit-core",
+    sharedSettings,
+    Compile / unmanagedSourceDirectories ++=
+      crossBuildingDirectories("munit-core", "main").value,
+    libraryDependencies ++= List(
+      "org.scala-lang" % "scala-reflect" % {
+        if (isScala3Setting.value) scala213
+        else scalaVersion.value
+      } % Provided
+    )
+  )
+  .jvmSettings(
+    sharedJVMSettings
+  )
+  .nativeConfigure(sharedNativeConfigure)
+  .nativeSettings(
+    sharedNativeSettings
+  )
+  .jsConfigure(sharedJSConfigure)
+  .jsSettings(sharedJSSettings)
+
+lazy val munitDiff = crossProject(JSPlatform, JVMPlatform, NativePlatform)
```

Comment:
Ideally munitDiff would not depend on anything, it should not be about testing really.

Reply:
As far as I understand it, both diff and munit need things like `Location` along with macros that generate it. Do you prefer code duplication over shared core module?

Reply:
I think we can try getting rid of `Location`, I'll give it a try. What about `AnsiColors`? Should it be a part of diff module?

Reply:
I think so, since it might be useful for showing the diff.